File: server/src/main/java/com/ivantrykosh/app/budgettracker/server/dtos/AccountDto.java
package com.ivantrykosh.app.budgettracker.server.dtos;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

/**
 * Account dto
 */
@NoArgsConstructor
@Getter
@Setter
@ToString
public class AccountDto {
    private Long accountId; // Account ID

    private Long userId; // ID of user to whom account belongs

    private String name; // Name of account

    private Double incomesSum; // Sum of incomes

    private Double expensesSum; // Sum of expenses
}

File: client/app/src/main/java/com/ivantrykosh/app/budgettracker/client/data/remote/dto/AccountDto.kt
package com.ivantrykosh.app.budgettracker.client.data.remote.dto

import com.ivantrykosh.app.budgettracker.client.domain.model.Account

/**
 * Data class for account
 */
data class AccountDto(
    val accountId: Long,
    val userId: Long,
    val name: String,
    val incomesSum: Double,
    val expensesSum: Double
)

/**
 * Convert from AccountDto to Account
 */
fun AccountDto.toAccount(): Account {
    return Account(
        accountId = accountId,
        userId = userId,
        name = name,
        total = incomesSum + expensesSum
    )
}